docker/build-image.sh

```diff
@@ -23,25 +23,58 @@ echo ""
 echo "##### CARMA Docker Image Build Script #####"
 echo ""
 
-if [[ -z "$1" ]]; then
-    FULL_VERSION_STRING=$("../engineering_tools/get-carma-version.sh")
-else
-    FULL_VERSION_STRING="$1"
+while [[ $# -gt 0 ]]; do
+    arg="$1"
+    case $arg in
+        -v|--version)
+            COMPONENT_VERSION_STRING="$2"
+            shift
+            shift
+            ;;
+        --system-release)
+            SYSTEM_RELEASE=true
+            shift
+            ;;
+        -p|--push)
+            PUSH=true
+            shift
+            ;;
+    esac
+done
+
+if [[ -z "$COMPONENT_VERSION_STRING" ]]; then
+    COMPONENT_VERSION_STRING=$("./get-component-version.sh")
 fi
 
-echo "Building docker image for CARMA version: $FULL_VERSION_STRING"
-echo "Final image name: $USERNAME/$IMAGE:$FULL_VERSION_STRING"
+echo "Building docker image for CARMA version: $COMPONENT_VERSION_STRING"
+echo "Final image name: $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING"
 
 cd ..
-docker build --no-cache -t $USERNAME/$IMAGE:$FULL_VERSION_STRING \
-    --build-arg VERSION="$FULL_VERSION_STRING" \
+docker build --no-cache -t $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING \
+    --build-arg VERSION="$COMPONENT_VERSION_STRING" \
     --build-arg VCS_REF=`git rev-parse --short HEAD` \
     --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` .
 
-docker tag $USERNAME/$IMAGE:$FULL_VERSION_STRING $USERNAME/$IMAGE:latest
+TAGS=()
+TAGS+=("$USERNAME/$IMAGE:$COMPONENT_VERSION_STRING")
+
+docker tag $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING $USERNAME/$IMAGE:latest
+TAGS+=("$USERNAME/$IMAGE:latest")
 
-echo "Tagged $USERNAME/$IMAGE:$FULL_VERSION_STRING as $USERNAME/$IMAGE:latest"
+echo "Tagged $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING as $USERNAME/$IMAGE:latest"
+
+if [ "$SYSTEM_RELEASE" = true ]; then
+    SYSTEM_VERSION_STRING=$("./get-system-version.sh")
+    docker tag $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING $USERNAME/$IMAGE:$SYSTEM_VERSION_STRING
+    echo "Tagged $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING as $USERNAME/$IMAGE:$SYSTEM_VERSION_STRING"
```

Comment:
Does this redundantly apply the component tag? It seems this was already handled on line 61 above.

Reply:
The tag command is `docker tag <CURRENT_NAME> <NEW_ADDITIONAL_NAME>` so in this case it references the image by the component tag, then also assigns it the `$SYSTEM_VERSION_STRING` tag too.